### ISSUE TITLE
automatic Float / support creation

### ISSUE THREAD

**Issue by Aslan0** [entropybit/slaprinter#10]

**Comment by entropybit** [comment]
What exactly is meant by automatic Float / support creation ?

**Comment by Aslan0** [comment]
Floats are platforms that are created at the very first layer.... as a support platform for the build. the connection to the standard build needs to be really thin so it can be broken off easily. floats are necessary for us in case normal prints wont stick to the platform like theyre supposed to.

supports are additional structures that are added before slicing. they resemble pillars and make sure the structure is stable.

filling is the third one of those added structures: we have to have an algorithm that fills hollow spaces with a for example hexagonal structure to give the build stability. 

if you still dont know what im referring to or want to see how it works in action, id recommend printing something on the fablab 3d printers. Itll become clear then.

**Comment by entropybit** [comment]
Ah I think i get it so you essentially have to add something to garanty a good grip to the plattform which pulls out the printed layers if the structure has no large flat sides, like a ball or something like that.
I think that could be done by a algorithm working like ray tracing from the support platform to the largest slice to determine if and where to add fills / floats ? (hopefully this is descriiption is clear enough on what I mean)

**Comment by Aslan0** [comment]
Yeah, something like that. And the basic platform (=float) as well as the supports (=pillars connecting object and float) have to be able to be turned off and on by the user (thats what the other programs offer at least.)
